src/localization/database/cost_matrix.h

```diff
@@ -0,0 +1,45 @@
+#ifndef SRC_DATABASE_COST_MATRIX_H_
+#define SRC_DATABASE_COST_MATRIX_H_
+
+#include "features/feature_factory.h"
+
+#include <string>
+#include <vector>
+
+using Matrix = std::vector<std::vector<double>>;
```

Comment:
This should not be outside the class. This pollutes the global namespace for all the files that include this file. It would be better to move it into your `CostMatrix` class below.